lib/jsdom/living/events/MouseEvent.webidl

```diff
@@ -52,3 +51,27 @@ partial interface MouseEvent {
                         optional short buttonArg = 0,
                         optional EventTarget? relatedTargetArg = null);
 };
+
+// https://www.w3.org/TR/cssom-view/#extensions-to-the-mouseevent-interface
+// Adds attributes and changes existing ones to doubles
+partial interface MouseEvent {
+  readonly attribute double screenX;
+  readonly attribute double screenY;
+  readonly attribute double pageX;
+  readonly attribute double pageY;
+  readonly attribute double clientX;
+  readonly attribute double clientY;
+  readonly attribute double x;
+  readonly attribute double y;
+  readonly attribute double offsetX;
+  readonly attribute double offsetY;
+};
+
+// https://www.w3.org/TR/cssom-view/#extensions-to-the-mouseevent-interface
+// Changes existing coordinate entries to doubles
```

Comment:
I'm not sure how well this works in webidl2js. I guess it didn't break the build, which is good, but does it properly "override" the existing ones?

I think before merging, I need a test to ensure that these correctly come out as doubles instead of longs. So, construct a new MouseEvent with its screenX/screenY/clientX/clientY constructor options set to something fractional (e.g. `1.5`), and then read off the results from the object and ensure they're still fractional.

Bonus points if you add a test that using initMouseEvent does *not* preserve the fractionalness, since the arguments are still defined to be longs.

Reply:
Interestingly, it seems that although major browsers (chromium/firefox/webkit) have implemented these MouseEvent properties, they still convert doubles to longs (possibly for [legacy compatibility reasons](https://w3c.github.io/pointerevents/#the-click-auxclick-and-contextmenu-events)?) I don't know if that means jsdom **_shouldn't_** support doubles, but perhaps something to think about.

Reply:
Added tests around this